### PR TITLE
Added new resource SiteDesignRights

### PR DESCRIPTION
**Pull request by desmay**
#### Pull Request (PR) description
Add site design rights 

#### This Pull Request (PR) fixes the following issues
New resources

### CODE REVIEW

File: Modules/Office365DSC/DSCResources/MSFT_SPOSiteDesignRights/readme.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+## Description
+
+This resource configures rights on Site Designs.
+
+## Parameters
+
+    Ensure
+      - Required: No (Defaults to 'Present')
+      - Description: `Present` is the only value accepted.
+          Configurations using `Ensure = 'Absent'` will throw an Error!
```
Comment (NikCharlebois) [inline Modules/Office365DSC/DSCResources/MSFT_SPOSiteDesignRights/readme.md]: Is it? Looks like you are properly handling the Absent case.

File: Modules/Office365DSC/DSCResources/MSFT_SPOSiteDesignRights/MSFT_SPOSiteDesignRights.psm1 [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+        Write-Verbose -Message "Revoking SiteDesign rights on  $UserPrincipals for site design $SiteDesignTitle"
+        Revoke-PnPSiteDesignRights -Identity $cursiteDesign.Id -Principals $UserPrincipals
+    }
+
+    #No site design rights currently exist so add them
+    If ($currentSiteDesignRights.Ensure -eq "Absent")
+    {
+        Write-Verbose -Message "Granting SiteDesign rights on site design $SiteDesignTitle"
+        Grant-PnPSiteDesignRights -Identity $cursiteDesign.Id -Principals $UserPrincipals -Rights $Rights
+
```
Comment (NikCharlebois) [inline Modules/Office365DSC/DSCResources/MSFT_SPOSiteDesignRights/MSFT_SPOSiteDesignRights.psm1]: Remove empty line

File: Modules/Office365DSC/DSCResources/MSFT_SPOSiteDesignRights/readme.md (hunk 1 above)
Comment (desmay) [reply Modules/Office365DSC/DSCResources/MSFT_SPOSiteDesignRights/readme.md]: fixed

File: Modules/Office365DSC/DSCResources/MSFT_SPOSiteDesignRights/MSFT_SPOSiteDesignRights.psm1 (hunk 2 above)
Comment (desmay) [reply Modules/Office365DSC/DSCResources/MSFT_SPOSiteDesignRights/MSFT_SPOSiteDesignRights.psm1]: done

**Pull request merged**